src/reconcile/rec_write.c

```diff
@@ -3057,8 +3068,8 @@ __rec_split_discard(WT_SESSION_IMPL *session, WT_PAGE *page)
          * confirm backing blocks we care about, and free any disk image/saved updates.
          */
         if (multi->addr.block_cookie != NULL) {
-            WT_RET(__wt_btree_block_free(
-              session, multi->addr.block_cookie, multi->addr.block_cookie_size, false));
+            WT_RET(__wt_btree_block_free(session, multi->addr.block_cookie,
+              multi->addr.block_cookie_size, r->multi_next == 1));
```

Comment:
I'm not sure if we should free this block or not if this page is reconciled the first time. If we don't free it and the node crash, this page will never be included in any checkpoint. Can the page service detect this and free it?